### ISSUE TITLE
Python version conflict between CogVLM2 dependencies and CogNav testing environment

### ISSUE THREAD

**Issue by novada0** [yhanCao/CogNav_ObjNav#5]
In the step “Install CogVLM3 Please checkout CogVLM2 to install the dependencies”, the documentation at https://github.com/zai-org/CogVLM2/blob/main/basic_demo/README_zh.md
 mentions that Python needs to be version 3.10.12 or higher. However, your project was tested only with Python 3.8 on Ubuntu 22.04. Could you please clarify which Python version is officially recommended for CogVLM3 installation?

**Comment by yhanCao** [comment]
We only test CogNav with CogVLM2, in python version 3.8 and it works. Now I have a conda environment with the newest python version, the cognitive map model can work well, too. So  I think the python version does not affect the code execution.

**Comment by novada0** [comment]
Sorry to bother you again! I still encountered some issues while trying to reproduce CogNav and would like to ask for your guidance:
  In a Python 3.8 environment, I ran “pip install -r environment.txt” to install the dependencies, but it reported that visualization_msgs could not be found. I understand that visualization_msgs is not a standard Python package on PyPI. Could you please advise if I need to install ROS first in order to use it?

<img width="1280" height="346" alt="Image" src="https://github.com/user-attachments/assets/c252bfb8-0b33-4934-9f93-13a460d62e33" />

  Later, I switched to Python 3.10 and configured the GradSLAM environment, but open3d==0.10.0.0 requires Python 3.8, which caused a version conflict too.
  To resolve these environment issues, would it be possible for you to share the complete list of package versions or the exported environment used in your experiments?
  Thank you very much for your assistance!

**Comment by yhanCao** [comment]
Here is my requirement.txt
absl-py==2.2.2
actionlib==1.14.0
addict==2.4.0
aiofiles==23.2.1
algo==1.3.1
angles==1.9.13
annotated-types==0.7.0
antlr4-python3-runtime==4.9.3
anyio==4.5.2
appdirs==1.4.4
asttokens==3.0.0
astunparse==1.6.3
attrs==25.3.0
azure-core==1.33.0
azure-identity==1.21.0
azure-storage-blob==12.25.1
backcall==0.2.0
base_local_planner==1.17.3
black==24.8.0
blinker==1.8.2
bondpy==1.8.6
cachetools==5.5.2
camera-calibration==1.17.0
camera-calibration-parsers==1.12.0
catkin==0.8.10
certifi==2025.4.26
cffi==1.17.1
chamferdist==1.0.3
charset-normalizer==3.4.1
cityscapesScripts==2.2.4
click==8.1.8
clip==0.2.0
cloudpickle==3.1.1
coloredlogs==15.0.1
comm==0.2.2
ConfigArgParse==1.7
contourpy==1.1.1
controller-manager==0.20.0
controller-manager-msgs==0.20.0
cryptography==45.0.2
cv-bridge==1.16.2
cycler==0.12.1
dash==3.0.4
decorator==5.2.1
defusedxml==0.7.1
-e git+https://github.com/facebookresearch/detectron2.git@65184fc057d4fab080a98564f6b60fae0b94edc4#egg=detectron2
diagnostic-analysis==1.11.0
diagnostic-common-diagnostics==1.11.0
diagnostic-updater==1.11.0
diffdist==0.1
distinctipy==1.3.4
distro==1.9.0
docker-pycreds==0.4.0
docstring_parser==0.16
dynamic-reconfigure==1.7.3
einops==0.8.1
eval_type_backport==0.2.2
exceptiongroup==1.3.0
executing==2.2.0
faiss-gpu==1.7.2
fastapi==0.115.12
fastjsonschema==2.21.1
ffmpy==0.5.0
filelock==3.16.1
Flask==3.0.3
flatbuffers==25.2.10
fonttools==4.57.0
fsspec==2025.3.0
ftfy==6.2.3
fvcore==0.1.5.post20221221
gast==0.4.0
gazebo_plugins==2.9.2
gazebo_ros==2.9.2
gencpp==0.7.0
geneus==3.0.0
genlisp==0.4.18
genmsg==0.6.0
gennodejs==2.0.2
genpy==0.6.15
gitdb==4.0.12
GitPython==3.1.44
google-auth==2.40.2
google-auth-oauthlib==1.0.0
google-pasta==0.2.0
gradio==4.44.1
gradio_client==1.3.0
-e git+https://github.com/gradslam/gradslam.git@44470eee4484aaad703d7bf962a8f42496867a9d#egg=gradslam
grpcio==1.70.0
gym==0.26.2
gym-notices==0.0.8
h11==0.16.0
h5py==3.11.0
-e git+https://github.com/facebookresearch/habitat-lab.git@083f62da8113b6dd93b1d3b6515dfe9431c70c9e#egg=habitat
habitat_sim==0.2.1
hf-xet==1.1.2
-e git+https://github.com/hovsg/HOV-SG.git@6d9cd24d7d3b877896b00b7f1c97c2be86cdfa04#egg=hovsg
httpcore==1.0.9
httpx==0.28.1
huggingface-hub==0.32.0
humanfriendly==10.0
hydra-core==1.3.2
idna==3.10
ifcfg==0.24
image-geometry==1.16.2
imageio==2.35.1
imageio-ffmpeg==0.5.1
importlib_metadata==8.5.0
importlib_resources==6.4.5
interactive-markers==1.12.0
iopath==0.1.9
ipython==8.12.3
ipywidgets==8.1.7
isodate==0.7.2
itsdangerous==2.2.0
jedi==0.19.2
Jinja2==3.1.6
jiter==0.9.1
joblib==1.4.2
joint-state-publisher==1.15.1
joint-state-publisher-gui==1.15.1
json-tricks==3.17.3
jsonschema==4.23.0
jsonschema-specifications==2023.12.1
jupyter_core==5.7.2
jupyterlab_widgets==3.0.15
keras==2.13.1
kiwisolver==1.4.7
kornia==0.7.3
kornia_rs==0.1.9
laser_geometry==1.6.7
lazy_loader==0.4
libclang==18.1.1
lightning-utilities==0.11.9
llvmlite==0.41.1
magnum @ file:///home/javens/3rd_party/habitat-sim/build/deps/magnum-bindings/src/python
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib==3.7.5
matplotlib-inline==0.1.7
mdurl==0.1.2
message-filters==1.17.0
modelscope==1.24.1
mpi4py==4.0.3
mpmath==1.3.0
msal==1.32.3
msal-extensions==1.3.0
MultiScaleDeformableAttention @ file:///home/javens/3rd_party/Deformable-DETR/models/ops
mup==1.0.0
mypy_extensions==1.1.0
narwhals==1.40.0
natsort==8.4.0
nbformat==5.10.4
nest-asyncio==1.6.0
networkx==3.1
nltk==3.9.1
numba==0.58.1
numpy==1.23.4
numpy-quaternion==2023.0.4
nvidia-cublas-cu11==11.11.3.6
nvidia-cuda-cupti-cu11==11.8.87
nvidia-cuda-nvrtc-cu11==11.8.89
nvidia-cuda-runtime-cu11==11.8.89
nvidia-cudnn-cu11==8.7.0.84
nvidia-cufft-cu11==10.9.0.58
nvidia-curand-cu11==10.3.0.86
nvidia-cusolver-cu11==11.4.1.48
nvidia-cusparse-cu11==11.7.5.86
nvidia-nccl-cu11==2.20.5
nvidia-nvtx-cu11==11.8.86
oauthlib==3.2.2
omegaconf==2.3.0
open3d==0.18.0
open_clip_torch==2.32.0
openai==1.82.0
opencv-python==4.11.0.86
opencv-python-headless==4.11.0.86
opt_einsum==3.4.0
orjson==3.10.15
packaging==25.0
pandas==2.0.3
panopticapi @ git+https://github.com/cocodataset/panopticapi.git@7bb4655548f98f3fedc07bf37e9040a992b054b0
parso==0.8.4
pathspec==0.12.1
pexpect==4.9.0
pickleshare==0.7.5
Pillow==9.5.0
pkgutil_resolve_name==1.3.10
platformdirs==4.3.6
plotly==6.1.1
plyfile==1.0.3
pooch==1.8.2
portalocker==3.0.0
prompt_toolkit==3.0.51
protobuf==4.25.7
psutil==7.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
pyarrow==17.0.0
pyasn1==0.6.1
pyasn1_modules==0.4.2
pycocotools==2.0.7
pycparser==2.22
pydantic==2.10.6
pydantic_core==2.27.2
pydub==0.25.1
Pygments==2.19.1
PyJWT==2.9.0
pyparsing==3.1.4
pyquaternion==0.9.9
python-dateutil==2.9.0.post0
python-multipart==0.0.20
python-qt-binding==0.4.4
-e git+https://github.com/facebookresearch/pytorch3d.git@2f11ddc5ee7d6bd56f2fb6744a16776fab6536f7#egg=pytorch3d
pytz==2025.2
pyvista==0.44.2
PyWavelets==1.4.1
PyYAML==6.0.2
qt-dotgraph==0.4.2
qt-gui==0.4.2
qt-gui-cpp==0.4.2
qt-gui-py-common==0.4.2
referencing==0.35.1
regex==2024.11.6
requests==2.32.3
requests-oauthlib==2.0.0
resource_retriever==1.12.7
retrying==1.3.4
rich==14.0.0
rosbag==1.17.0
rosboost-cfg==1.15.8
rosclean==1.15.8
roscreate==1.15.8
rosgraph==1.17.0
roslaunch==1.17.0
roslib==1.15.8
roslint==0.12.0
roslz4==1.17.0
rosmake==1.15.8
rosmaster==1.17.0
rosmsg==1.17.0
rosnode==1.17.0
rosparam==1.17.0
rospy==1.17.0
rosservice==1.17.0
rostest==1.17.0
rostopic==1.17.0
rosunit==1.15.8
roswtf==1.17.0
rpds-py==0.20.1
rqt-console==0.4.12
rqt-image-view==0.4.17
rqt-logger-level==0.4.12
rqt-moveit==0.5.11
rqt-reconfigure==0.5.5
rqt-robot-dashboard==0.5.8
rqt-robot-monitor==0.5.15
rqt-runtime-monitor==0.5.10
rqt-rviz==0.7.0
rqt-tf-tree==0.6.4
rqt_action==0.4.9
rqt_bag==0.5.1
rqt_bag_plugins==0.5.1
rqt_dep==0.4.12
rqt_graph==0.4.14
rqt_gui==0.5.3
rqt_gui_py==0.5.3
rqt_launch==0.4.9
rqt_msg==0.4.10
rqt_nav_view==0.5.7
rqt_plot==0.4.13
rqt_pose_view==0.5.11
rqt_publisher==0.4.10
rqt_py_common==0.5.3
rqt_py_console==0.4.10
rqt_robot_steering==0.5.12
rqt_service_caller==0.4.10
rqt_shell==0.4.11
rqt_srv==0.4.9
rqt_top==0.4.10
rqt_topic==0.4.13
rqt_web==0.4.10
rsa==4.9.1
ruff==0.11.11
rviz==1.14.25
safetensors==0.5.3
scikit-fmm==2024.5.29
scikit-image==0.21.0
scikit-learn==1.3.2
scipy==1.10.0
scooby==0.10.1
seaborn==0.13.2
segment_anything @ git+https://github.com/facebookresearch/segment-anything.git@dca509fe793f601edb92606367a655c15ac00fdf
semantic-version==2.10.0
sensor-msgs==1.13.1
sentencepiece==0.2.0
sentry-sdk==2.29.1
setproctitle==1.3.6
shapely==2.0.7
shellingham==1.5.4
shtab==1.7.2
six==1.17.0
smach==2.5.2
smach-ros==2.5.2
smclib==1.8.6
smmap==5.0.2
sniffio==1.3.1
stack-data==0.6.3
starlette==0.44.0
supervision==0.13.0
sympy==1.13.3
tabulate==0.9.0
tenacity==9.0.0
tensorboard==2.13.0
tensorboard-data-server==0.7.2
tensorflow==2.13.1
tensorflow-estimator==2.13.0
tensorflow-io-gcs-filesystem==0.34.0
termcolor==2.4.0
tf==1.13.2
tf-conversions==1.13.2
tf2-geometry-msgs==0.7.7
tf2-kdl==0.7.7
tf2-py==0.7.7
tf2-ros==0.7.7
threadpoolctl==3.5.0
tifffile==2023.7.10
timm==0.4.12
tokenizers==0.20.3
tomli==2.2.1
tomlkit==0.12.0
topic-tools==1.17.0
torch @ file:///media/javens/rainbow7/3rd_party/torch_wheel/torch-2.3.1-cu118-py3.8/torch-2.3.1%2Bcu118-cp38-cp38-linux_x86_64.whl#sha256=5669916fed356a6a4034aeaf9c78184bd1b4467b06d75d95f6540dd16969ad31
torchaudio @ file:///media/javens/rainbow7/3rd_party/torch_wheel/torch-2.3.1-cu118-py3.8/torchaudio-2.3.1%2Bcu118-cp38-cp38-linux_x86_64.whl#sha256=b7f50b6b13478f98af7ee9c4a50ad28ce9041364b47316012216c542ee734b77
torchmetrics==1.5.2
torchvision @ file:///media/javens/rainbow7/3rd_party/torch_wheel/torch-2.3.1-cu118-py3.8/torchvision-0.18.1%2Bcu118-cp38-cp38-linux_x86_64.whl#sha256=38437061fc8652d9cc88018051379b5fd623b1d6e02a3e36df01817277871799
tqdm==4.67.1
traitlets==5.14.3
transformers==4.46.3
trimesh==4.6.10
triton==2.3.1
typeguard==4.4.0
typer==0.15.4
typing==3.7.4.3
typing_extensions==4.13.2
tyro==0.9.21
tzdata==2025.2
urllib3==2.2.3
uvicorn==0.33.0
vision-datasets==0.2.2
vtk==9.3.1
wandb==0.19.11
watchfiles==0.24.0
wcwidth==0.2.13
websockets==12.0
Werkzeug==3.0.6
widgetsnbextension==4.0.14
wrapt==1.17.2
xacro==1.14.18
yacs==0.1.8
zipp==3.20.2

**Comment by yhanCao** [comment]
> Sorry to bother you again! I still encountered some issues while trying to reproduce CogNav and would like to ask for your guidance: In a Python 3.8 environment, I ran “pip install -r environment.txt” to install the dependencies, but it reported that visualization_msgs could not be found. I understand that visualization_msgs is not a standard Python package on PyPI. Could you please advise if I need to install ROS first in order to use it?
> 
> <img alt="Image" width="1280" height="346" src="https://private-user-images.githubusercontent.com/107106546/481904363-c252bfb8-0b33-4934-9f93-13a460d62e33.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyNzQ1MDIsIm5iZiI6MTc1NjI3NDIwMiwicGF0aCI6Ii8xMDcxMDY1NDYvNDgxOTA0MzYzLWMyNTJiZmI4LTBiMzMtNDkzNC05ZjkzLTEzYTQ2MGQ2MmUzMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyN1QwNTU2NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MThhN2I1MzQ4YjIxYmRlNmE3NTY1ZTMyYTlhZmIxZDM2YjdmODE4MDBlZjg0MjM4ZTVmYTdkNzk5ZmY4NTkxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.XZNK4ItEAElyCc4uskYhZR9UsKByLvVY7WhHKY0wYVA">
> Later, I switched to Python 3.10 and configured the GradSLAM environment, but open3d==0.10.0.0 requires Python 3.8, which caused a version conflict too. To resolve these environment issues, would it be possible for you to share the complete list of package versions or the exported environment used in your experiments? Thank you very much for your assistance!

If you use the python 3.10, use these to install gradslam:
git clone https://github.com/gradslam/gradslam.git
cd gradslam
git checkout conceptfusion
pip install .
